src/config.rs

```diff
@@ -68,6 +73,69 @@ impl Config {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct SubProcess {
+    pub command: String,
+    pub working_dir: Option<PathBuf>,
+    pub foreground: bool,
+    #[serde(default)]
+    pub keep_on_exit: bool,
+    pub stdout: Option<bool>,
+    pub stderr: Option<bool>,
+}
+
+impl SubProcess {
+    pub fn spawn(&self, tmp_dir: &Path) -> Option<process::Child> {
+        let mut process = process::Command::new(run::get_shell());
+
+        process.current_dir(self.working_dir.as_ref().unwrap_or(&tmp_dir.to_path_buf()));
```

Comment:
Use `as_deref()` instead to get a `&Path` so you don't need to convert tmp_dir to a pathbuf